markdown_it/rules_core/replacements.py

```diff
@@ -0,0 +1,100 @@
+"""Simple typographic replacements
+
+(c) (C) → ©
+(tm) (TM) → ™
+(r) (R) → ®
++- → ±
+(p) (P) -> §
+... → … (also ?.... → ?.., !.... → !..)
+???????? → ???, !!!!! → !!!, `,,` → `,`
+-- → &ndash;, --- → &mdash;
+"""
+import logging
+import re
+from typing import List
+
+from .state_core import StateCore
+from ..token import Token
+
+LOGGER = logging.getLogger(__name__)
+
+# TODO:
+# - fractionals 1/2, 1/4, 3/4 -> ½, ¼, ¾
+# - miltiplication 2 x 4 -> 2 × 4
+
+RARE_RE = r"\+-|\.\.|\?\?\?\?|!!!!|,,|--"
```

Comment:
```suggestion
RARE_RE = re.compile(r"\+-|\.\.|\?\?\?\?|!!!!|,,|--")
```

It's a bit more performant to pre-compile, then use e.g. `RARE_RE.search`